### PR TITLE
[controller] First cleanup of CoreBluetoothMgr.py

### LINKED ISSUE #4183: [controller] Python Device Controller code need cleanup
I would also complain about this should_tuple (what is it?). From what I see, it would be:

```python
(command, start_time, timeout) = should_tuple
```

but in the code I see an instance where the tuple has 4(!) values.

Overall this python code needs cleanup. There should be no tuple, but rather a class. It would read much easier like::

```python
def runLoopUntil(self, condition):
   if codition.Expired():
     return False

   if condition.command == "ready":
      return not self.ready_condition
   # .....
```

not sure when the right time for cleanup would be, however if this code is to be updated in time, I think we should spend some time cleaning it up before adding more features to it.

_Originally posted by @andy31415 in https://github.com/project-chip/connectedhomeip/pull/4175#discussion_r541072655_

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
I would also complain about this should_tuple (what is it?). From what I see, it would be:

(command, start_time, timeout) = should_tuple
but in the code I see an instance where the tuple has 4(!) values.

Overall this python code needs cleanup. There should be no tuple, but rather a class. It would read much easier like::

def runLoopUntil(self, condition):
   if codition.Expired():
     return False

   if condition.command == "ready":
      return not self.ready_condition
   # .....
not sure when the right time for cleanup would be, however if this code is to be updated in time, I think we should spend some time cleaning it up before adding more features to it.

Originally posted by @andy31415 in #4175 (comment)

 #### Summary of Changes

Add `BgLoopCondition` to replace `should_tuple`

 Fixes #4183

### CODE REVIEW

File: src/controller/python/chip/ChipCoreBluetoothMgr.py [hunk 1]
```diff
@@ -230,43 +239,41 @@ def setInputHook(self, hookFunc):
...
+        elif cond.op == "unsubscribe":
             return self.subscribe_condition
 
-        return False
+        return True
 
-    def runLoopUntil(self, should_tuple):
+    def runLoopUntil(self, cond):
         """Helper function to drive OSX runloop until an expected event is received or
         the timeout expires."""
```
Comment (bzbarsky-apple) [inline src/controller/python/chip/ChipCoreBluetoothMgr.py:272]: Maybe document that `cond` is a `BgLoopCondition`?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/python/chip/ChipCoreBluetoothMgr.py (hunk 1 above)
Comment (erjiaqing) [reply src/controller/python/chip/ChipCoreBluetoothMgr.py:275]: Done

**Review by andy31415 — APPROVED** [review]
Thank you!

File: src/controller/python/chip/ChipCoreBluetoothMgr.py [hunk 2]
```diff
@@ -106,6 +106,15 @@ def _VoidPtrToCBUUID(ptr, len):
 
     return ptr
 
+class BgLoopCondition:
```
Comment (mspang) [inline src/controller/python/chip/ChipCoreBluetoothMgr.py:109]: Bg is "Background" ?

We should abbreviating if it creates uncertainty and just say "Background". Or just remove "Bg" altogether if it's not carrying it's weight.

**Review by mspang — COMMENTED** [review]
Can you fill in the PR template ?

File: src/controller/python/chip/ChipCoreBluetoothMgr.py (hunk 2 above)
Comment (erjiaqing) [reply src/controller/python/chip/ChipCoreBluetoothMgr.py]: Done

**Review by rwalker-apple — APPROVED** [review]

**Comment by rwalker-apple** [comment]
@jelderton @BroderickCarlin @saurabhst @hawk248 ?

**Review by mspang — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]
thank you!

**Comment by yunhanw-google** [comment]
@BroderickCarlin 
@saurabhst

**Review by saurabhst — APPROVED** [review]

**Pull request merged**